### PR TITLE
#20777 workflow email improvements

### LINKED ISSUE #20777: Send email "From" the config portlet Email Address rather than using anonymous user for workflow emails
**Describe the bug**
The addresses anonymous@dotcmsfakeemail.org and system@dotcmsfakeemail.org are hard-coded. These addresses are used to send email and therefore need to be configurable to ensure delivery of automated emails from dotCMS.

In many situations, email messages from these addresses will be rejected by SMTP servers, or will be categorized as Spam.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This comes to solve a problem with the workflow email actionlets.. we're currently sending emails to the workflow assignee but in many cases, this results in anonymous or system users. for which the resulting emails were coming from a fake email address that might result in the email getting caught by the spam filter. We're relying on the company email address to send those emails. Also, an additional fallback has been added in case the company lacks a valid e-mail address.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/system/CompanyEmailForm.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import javax.validation.constraints.NotNull;
+import com.dotcms.rest.api.Validated;
+
+public class CompanyEmailForm extends Validated {
+
+    @NotNull
+    private final String email;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/system/CompanyEmailForm.java:11]: isn't a annotation for email?

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/system/CompanyEmailForm.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/system/CompanyEmailForm.java:11]: it's validated via Apache utils

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/curl-test/ConfigurationResource.postman_collection.json [hunk 2]
```diff
@@ -12,53 +12,197 @@
...
+					],
+					"path": [
+						"api",
+						"",
+						"v1",
+						"configuration",
+						"_validateCompanyEmail"
+					]
+				},
+				"description": "This tests the endpoint that brings back one specific App/integration given the App-key followed by the site-id"
```
Comment (nollymar) [inline dotCMS/src/curl-test/ConfigurationResource.postman_collection.json:135]: Is this description valid?

Comment (fabrizzio-dotCMS) [reply dotCMS/src/curl-test/ConfigurationResource.postman_collection.json:135]: no.. needs to be changed

File: dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties [hunk 3]
```diff
@@ -1888,6 +1888,9 @@ logout-as = Salir de
 Logout = Desloguear
 Lucene-Query = Query de Lucene
 mail-domain = Dominio de correo
+email-address-validate = Enviar correo de prueba
```
Comment (nollymar) [inline dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties:1891]: We shouldn't modify Language.properties for specific languages.

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/system/ConfigurationHelper.java [hunk 4]
```diff
@@ -272,5 +280,52 @@ private static String keyDigest(){
...
+	 * the result is communicated via system wide notification to the user
+	 * @param email
+	 * @return
+	 * @throws DotDataValidationException
+	 */
+	public void validateEmail(final String email, final User user)
+			throws ExecutionException, InterruptedException {
+
+		if (!EmailValidator.getInstance().isValid(email)) {
+			throw new IllegalArgumentException(
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/system/ConfigurationHelper.java:294]: Is there a test for this validation?

File: dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/WEB-INF/messages/Language_es.properties:1891]: why not? I always modify Eglish and Spanish

File: dotCMS/src/integration-test/resources/messages/Language_en_US.properties [hunk 5]
```diff
@@ -240,6 +240,9 @@ login-as=Login As
 logo=Logo
 logout-as=Logout As
 mail-domain=Mail Domain
+email-address-validate = Send test email
```
Comment (nollymar) [inline dotCMS/src/integration-test/resources/messages/Language_en_US.properties:243]: This file shouldn't be modified. Only Language.properties

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/system/ConfigurationHelper.java (hunk 4 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/system/ConfigurationHelper.java:294]: yes

**Review by nollymar — APPROVED** [review]

**Pull request merged**